cmd/terramate/cli/cli.go

```diff
@@ -738,6 +740,22 @@ func (s *kongParallelFlag) Decode(ctx *kong.DecodeContext) error {
 	return nil
 }
 
+type cloudSyncLayer string
+
+func (csl cloudSyncLayer) String() string {
+	return string(csl)
+}
+
+func (csl cloudSyncLayer) Validate() error {
+	for _, c := range csl {
+		if !unicode.IsLetter(c) && !unicode.IsDigit(c) && c != '-' {
```

Comment:
The format must be `[a-zA-Z]{1}[a-zA-Z0-9-]`. It cannot start with numbers or dash, otherwise it could create layers like `custom:1` or `custom:-` ...

Reply:
The layers are supposed to be used to represent terraform workspaces. I just tested the `terraform workspace new NAME` command and was able to create with the following names:

![image](https://github.com/terramate-io/terramate/assets/92498/64be9d5c-7cd7-424f-99cc-ca73ecac8bd1)


So even the alphanumerc validation in this PR is more restrictive than terraform, and with the regex that you suggested if a user has a workspace called `1` or `1-dev`, or `2-dev`, they wouldn't be able to use the previews feature.